### ISSUE TITLE
Training code?

### ISSUE THREAD

**Issue by nitinh12** [HiDream-ai/HiDream-I1#15]
Will you publish any training code?

**Comment by bghira** [comment]
i've made an implementation here: https://github.com/bghira/SimpleTuner/pull/1380

it's basically the same as Flux, except for the abstracted MoE bits, and the negative flow field (#14)

**Comment by caiqi** [comment]
Thanks for integrating this into SimpleTuner!

**Comment by bghira** [comment]
it is kind of not working yet. i have mostly AMD and Apple systems to test with, and flash_attn is in the way, but it'll probably work soon.

**Comment by bghira** [comment]
at this point it'll train around 40G VRAM minimum for 1024px batch size 1 Lycoris but there's some issues where the loss is pretty high around 1.0-2.5 @caiqi any ideas?

**Comment by linoytsaban** [comment]
I've noticed the same behavior with the loss ^ curious about it too

**Comment by roooobotx** [comment]
jjj
